### ISSUE TITLE
Imported playbooks not found/linted anymore.

### ISSUE THREAD

**Issue by TimGrt** [ansible/ansible-lint#3407]
##### Summary

A *global* playbook with only `import_playbook` statements is identified as generic yaml file instead of playbook. Bug was introduced in 6.12.0, previous versions (<6.11.0) are working. Bug does also occur in latest 6.15.0!

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 6.12.0 using ansible 2.14.4
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```console (paste below)
.
├── playbooks
│   ├── playbook1.yml
│   └── playbook2.yml
└── site.yml
```

```yaml
---
# site.yml
- ansible.builtin.import_playbook: playbooks/playbook1.yml
- ansible.builtin.import_playbook: playbooks/playbook2.yml
```
Both playbooks look like this:
```yaml
---
# playbooks/playbook1.yml
- hosts: localhost
  connection: local
  tasks:
    - debug:
        msg: "Hello from playbook1"
```

##### Desired Behavior

```console
$ ansible-lint --version
ansible-lint 6.11.0 using ansible 2.14.4
$ ansible-lint site.yml -vv
DEBUG    Logging initialized to level 10
...
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: site.yml
INFO     Executing syntax check on site.yml (0.24s)
DEBUG    Examining site.yml of type playbook
DEBUG    Examining playbooks/playbook2.yml of type playbook
DEBUG    Examining playbooks/playbook1.yml of type playbook
WARNING  Listing 8 violation(s) that are fatal
...
Failed after min profile: 4 failure(s), 4 warning(s) on 3 files.
A new release of ansible-lint is available: 6.11.0 → 6.15.0 Upgrade by running: pip3 install --user --upgrade ansible-lint
```

##### Actual Behavior

Running ansible-lint and providing `site.yml`  with Versions >=6.12.0 does not lint the imported playbooks, as the file is identified as a generic yaml file.

```console
$ ansible-lint site.yml -vv
DEBUG    Logging initialized to level 10
...
DEBUG    Passed 'site.yml' positional argument was identified as generic 'yaml' file kind.
...
Passed with production profile: 0 failure(s), 0 warning(s) on 1 files.
```

**Comment by ajinkyau** [comment]
@TimGrt Unable to reproduce this issue with the latest release of ansible-lint.
<pre>
ansible-lint --version => ansible-lint 6.15.0 using ansible 2.14.5.    
ansible-lint bug-3407 -vv
DEBUG    Passed 'bug-3407/site.yml' positional argument was identified as generic 'yaml' file kind.
INFO     Executing syntax check on playbook test/playbooks/playbook2.yml (0.29s)
INFO     Executing syntax check on playbook test/playbooks/playbook1.yml (0.29s)
DEBUG    Examining test/site.yml of type yaml
DEBUG    Examining test/playbooks/playbook2.yml of type playbook
DEBUG    Examining test/playbooks/playbook1.yml of type playbook
WARNING  Listing 6 violation(s) that are fatal
...
Failed after min profile: 6 failure(s), 0 warning(s) on 4 files.
</pre>
